tools/ddrgen/src/scanners/dwarf/AixDwarfParser.cpp

```diff
@@ -0,0 +1,2012 @@
+ *      http://www.opensource.org/licenses/apache2.0.php
+ *
+ * Contributors:
+ *    Multiple authors (IBM Corp.) - initial implementation and documentation
+ *******************************************************************************/
+
+#include "AixDwarfParser.hpp"
+/* Statics to create: */
+static file_map createdDies;
+static unordered_map<string, int> dieNames;
+/* This vector contains the file, typeID of the nested class, and parent class for populating nested classes */
+static vector<tuple<string, int, Dwarf_Die> > nestedClassesToPopulate;
+/* This vector contains the filename, typeID of the type reference, and type attribute to populate */
+static vector<tuple<string, int, Dwarf_Attribute> > refsToPopulate;
+/* This vector contains the Dwarf DIE with the bitfield attribute, which may have to be removed if it is unneeded */
+static vector<Dwarf_Die> bitFieldsToCheck;
+static Dwarf_Die _lastDie = NULL;
+static Dwarf_Die _builtInTypeDie = NULL;
+static Dwarf_Die _undefinedRef = NULL;
+/* _startNewFile is a state counter */
+static int _startNewFile = 0;
+/* Start numbering refs for refMap at one (zero will be for when no ref is found) */
+static Dwarf_Off refNumber = 1;
+/* This is used to initialize declLine to a unique value for each unique DIE, every time it is required */
+static Dwarf_Off declarationLine = 1;
+Dwarf_CU_Context * Dwarf_CU_Context::_firstCU;
+Dwarf_CU_Context * Dwarf_CU_Context::_currentCU;
+
+static void checkBitFields(Dwarf_Error *error);
+static string checkForDuplicateName();
+static void deleteDie(Dwarf_Die die);
+static int parseArray(const string data, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseBaseType(const string data, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseConstant(const string data, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseDwarfDie(const string data, Dwarf_Error *error);
+static int parseDwarfInfo(const char *line, Dwarf_Error *error);
+static Dwarf_Half parseDeclarationLine(string data, int *typeID, declFormat *format, string *declarationData, string *name);
+static int parseEnum(const string data, string dieName, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseFields(const string data, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseNewFormat(const string data, const string dieName, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseOldFormat(const string data, const string dieName, Dwarf_Die currentDie, Dwarf_Error *error);
+static int parseTypeDef(const string data, const string dieName, Dwarf_Die currentDie, Dwarf_Error *error);
+static void populateAttributeReferences();
+static int populateBuiltInTypeDies(Dwarf_Error *error);
+static void populateNestedClasses();
+static int setDieAttributes(const string dieName, const unsigned int dieSize, Dwarf_Die currentDie, Dwarf_Error *error);
+/* temporary debugging methods */
+using namespace std;
+void printType (Dwarf_Die * t) {
```

Comment:
You seem to have re-added these debug functions, possibly unintentionally? They use "cout" and do not follow the coding standards...

I suggest adding more DEBUGPRINTF's instead and print out the Die's as they are created. These prints can be turned on/off easily and as such can be left in instead of added and removed as you debug your code.